packages/dds/tree/src/feature-libraries/editable-tree/editableTree.ts

```diff
@@ -338,9 +340,31 @@ export interface EditableField
 	replaceNodes(index: number, newContent: ITreeCursor | ITreeCursor[], count?: number): void;
 }
 
+const editableTreeSlot = anchorSlot<EditableTree>();
+
 function makeTree(context: ProxyContext, cursor: ITreeSubscriptionCursor): EditableTree {
-	const newTarget = new NodeProxyTarget(context, cursor);
-	return adaptWithProxy(newTarget, nodeProxyHandler);
+	const anchor = cursor.buildAnchor() ?? fail("invalid");
+	const anchorNode = context.forest.anchors.locate(anchor) ?? fail("invalid");
+	const map = anchorNode.slotMap(editableTreeSlot);
+	const cached = map.get(editableTreeSlot);
+	if (cached !== undefined) {
+		context.forest.anchors.forget(anchor);
+		return cached;
+	}
+	const newTarget = new NodeProxyTarget(context, cursor, anchorNode, anchor);
+	const output = adaptWithProxy(newTarget, nodeProxyHandler);
+	map.set(editableTreeSlot, output);
+	anchorNode.on("afterDelete", cleanupTree);
+	map.get(editableTreeSlot);
+	return output;
+}
+
+function cleanupTree(anchor: AnchorNode): void {
+	const map = anchor.slotMap(editableTreeSlot);
+	const cached = map.get(editableTreeSlot);
+	if (cached !== undefined) {
```

Comment:
How could this be undefined?

Reply:
This can happen if the tree is disposed for some other reason (ex: its context is disposed) before this anchor is deleted.

I should prabably fix that that case leaks the event handler causing us to have to handle it.

Reply:
Fixed.